core/src/main/java/org/kiwix/kiwixmobile/core/utils/KiwixDialog.kt

```diff
@@ -110,16 +110,14 @@ sealed class KiwixDialog(
     R.string.did_you_know,
     R.string.hint_contents_drawer_message,
     R.string.got_it,
-    null,
-    icon = R.drawable.icon_question
+    null
```

Comment:
Could you please tell why did you remove icons from every dialog? I don't think that should be done. The dialogs which state clear history and clear all notes in settings should definitely have this icon. They should like alert dialogs. Also, opening an external link leads to the user exiting the app and opening a new link in an external browser. It also should most definitely have this exclamation icon to look like a proper warning. 

I think you were told to remove icons from just the delete bookmarks and delete history button in bookmarks and history activity as in that case the user is clicking on a trash bin icon. If the user clicks on a trash bin icon, the user knows it will lead to deletion and hence the exclamation icons need not be present. 

The other dialogs are made from clickable texts and not trash icons and hence they need to look like a warning. I think you should revert changes you have made here.

Reply:
@yashk2000 First of all, I was in favour of keeping icons.
The point is following material guidelines https://material.io/components/dialogs/#alert-dialog here. Please refer this first. So I thought to update my commit. I will revert id are required.

Reply:
I don't see it mentioning anywhere that using an icon is against md guidelines. We can most definitely use an icon in the places needed and ignore them where not needed.

Reply:
We should just keep in mind to not use unneeded action text.

Reply:
> It also should most definitely have this exclamation icon to look like a proper warning.
The dialogue itself is a proper warning

Reply:
> The dialogue itself is a proper warning

Yes, it is. But the icon provides for good UI as it will help capture the attention of the user effectively. Seeing the exclamation, I as a user would pay more attention to what the dialog is saying. 
I don't think there's any harm in keeping the icons at all. Why change something if it's already good unless the change results in something better?

Reply:
> The other dialogs are made from clickable texts and not trash icons and hence they need to look like a warning.

The clickable texts are obviously "texts" which are readable and which suggest the action so no point of difference between (icon and text) both suggest the action behind it.

>First of all, I was in favor of keeping icons.

I already said I was in favor of keeping icons.
This being a small issue involved quite a big discussion.
I'll do the changes as suggested by the mentors.

Reply:
> The clickable texts are obviously "texts" which are readable and which suggest the action so no point of difference between (icon and text) both suggest the action behind it.

What would you rather see? A simple icon that conveys something or read a large text? A simple image will convey a lot more than a text. 

> I'll do the changes as suggested by the mentors.

The changes were already suggested over [here](https://github.com/kiwix/kiwix-android/issues/1927#issuecomment-601762445).

> This being a small issue involved quite a big discussion.

Maybe your idea of removing the icons is actually better. The above comments were just my point of view :) 
The final decision, of course, rests with the maintainers.